### ISSUE TITLE
[v2 FEATURE]: sklearn pipeline integration

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#1075]
At the MLPDS meeting, I was asked if we could use a chemprop model like a sklearn estimator. We can't but they were interested in a wrapper to make it easier to integrate chemprop models into a sklearn pipeline.

**Comment by jxl26** [comment]
The limitation of the pipeline approach is that the transformer stage in which we incorporate MPNN produces fixed embeddings. This is not quite compatible with end-to-end training capability of the original model.

**Comment by KnathanM** [comment]
Yeah so it looks like we can't use a pipeline. I guess we need to pass all arguments for message passing, aggregation, and the FFN to the init of a single ChempropEstimator and have it construct the submodules instead of making the submodules first and passing those in. But the featurization transform (SMILES string to graph) and the ChempropEstimator can probably be pipelined together.

**Comment by JacksonBurns** [comment]
> But the featurization transform (SMILES string to graph) and the ChempropEstimator can probably be pipelined together.

I was able to do this flawlessly

I think the other application people would want would be to have the output of the chemprop estimator be the learned fingerprint - but that isn't trainable that way, and I'm not sure an easy way to incorporate it. So at a high level, I like the direction we are headed since it meshes the best with the scikit-learn world.
